src/netif.rs

```diff
@@ -873,20 +873,22 @@ where
 #[cfg(feature = "alloc")]
 mod driver {
     use core::borrow::Borrow;
+    use core::marker::PhantomData;
 
     use crate::handle::RawHandle;
     use crate::sys::*;
 
     use super::EspNetif;
 
-    pub struct EspNetifDriver<T>
+    pub struct EspNetifDriver<'d, T>
     where
         T: Borrow<EspNetif>,
     {
-        inner: alloc::boxed::Box<EspNetifDriverInner<T>>,
+        inner: alloc::boxed::Box<EspNetifDriverInner<'d, T>>,
+        _d: PhantomData<&'d ()>,
```

Comment:
This `PhantomData` is completely unnecessary, because another member - `EspNetifDriverInner` - is already lifetimed by `'d` so there is nothing "phantom" here. The phantom data is also incorrect - since the closure is `FnMut`, it should be should be `PhantomData<&'d mut ()>` instead.